haystack/pipelines/base.py

```diff
@@ -717,7 +717,12 @@ def save_to_yaml(self, path: Path, return_defaults: bool = False):
             component_type = component_instance.pipeline_config["type"]
             component_params = component_instance.pipeline_config["params"]
             components[node] = {"name": node, "type": component_type, "params": {}}
-            component_signature = inspect.signature(type(component_instance)).parameters
+            
+            component_parent_classes = inspect.getmro(type(component_instance))
```

Comment:
The name `component_parent_classes` could be a bit confusing as `getmro()` also returns the class itself. However, I don't have a better name and once one looks into what `getmro()` does it becomes obvious.